src/components/Territory.vue

```diff
@@ -109,19 +109,33 @@ export default {
 
     async checkIn() {
       if (window.confirm('Ready to check-in the territory?')) {
-        this.isLoading = true;
-        await this.resetNHRecords(this.id);
-        await this.getTerritory(this.id);
-        await this.checkinTerritory({
-          territoryId: this.id,
-          userId: get(this.territory, this.territory.status.publisher.id),
-          username: this.user.username,
-        });
-        this.isLoading = false;
-        await this.$router.push({ name: 'home' });
+        if (this.user.id === this.territory.status.publisher.id) {
+          this.isLoading = true;
+          await this.resetNHRecords(this.id);
+          await this.getTerritory(this.id);
+          await this.checkinTerritory({
+            territoryId: this.id,
+            userId: get(this.territory, this.territory.status.publisher.id),
+            username: this.user.username,
+          });
+          this.isLoading = false;
+          await this.$router.push({ name: 'home' });
+        } else {
+          // this.makeToast('danger');
```

Comment:
if a toast looks better, to introduce the warning, this is ready; it just needs to get un-commented. i didn't use it primarily because it feels like two separate systems at work: first a window.confirm, then a bootstrap toast. the alert(), which i ended up with seems like it works with the window.confirm.